File: LectureCodes/MatVec/Dense/mmstruc/Python/mmstruc.py
import numpy as np
from matplotlib import pyplot as plt


n = 100
A = np.diag(np.mgrid[:n])
A[:, -1] = np.mgrid[:n]
A[-1, :] = np.mgrid[:n]
B = A[::-1, :]
C = np.dot(A, A)
D = np.dot(A, B)

plt.figure()
plt.spy(A)
plt.savefig('../PYTHON_PICTURES/Aspy.eps')
plt.figure()
plt.spy(B)
plt.savefig('../PYTHON_PICTURES/Bspy.eps')
plt.figure()
plt.spy(C)
plt.savefig('../PYTHON_PICTURES/Cspy.eps')
plt.figure()
plt.spy(D)
plt.savefig('../PYTHON_PICTURES/Dspy.eps')
plt.show()

File: LectureCodes/MatVec/Dense/mmstruc/Matlab/mmstruc.m
n = 100; A = [diag(1:n-1), (1:n-1)'; (1:n) ]; B = A(n:-1:1,:);
C = A*A; D = A*B;
figure; spy(A,'r'); axis off; print -depsc2 '../PICTURES/Aspy.eps';
figure; spy(B,'r'); axis off; print -depsc2 '../PICTURES/Bspy.eps';
figure; spy(C,'r'); axis off; print -depsc2 '../PICTURES/Cspy.eps';
figure; spy(D,'r'); axis off; print -depsc2 '../PICTURES/Dspy.eps';
